source/blog/2023-11-14-hanami-210.html.markdown

```diff
@@ -0,0 +1,194 @@
+        def title_link
+          helpers.tag.a(title, href: context.routes.path(:post, id:))
+        end
+
+        def feature_box
+          render("feature_box", title: title, text: teaser_text)
+        end
+      end
+    end
+  end
+end
+```
+
+**Rendering views from actions is a breeze.** From the get go, actions render their matching view automatically, no extra work required. Once your views need certain input, you can also make that wiring clear:
+
+```ruby
+def handle(request, response)
+  response.render(view, id: request.params[:id])
+end
+```
+
+## Say hello to `app/assets/`
+
+With your views ready to go, it’s time to explore assets.
+
+Your assets live in `app/assets/`. JavaScript files live under `js/`, with `app` files serving as your entry points:
+
+```
+import "../css/app.css";
+
+console.log("Hello from app.ts");
+```
+
+As you can see, [TypeScript](https://www.typescriptlang.org) works out of the box. Just run `npm install typescript --save-dev`.
+
+## Your assets come fast and flexible
+
+**Hanami assets are powered by [esbuild](https://esbuild.github.io)**, giving you lightning quick build times.
+
+Modern front end affordances are ready for you out of the box, no configuration required, with our standard assets config a picture of simplicity:
+
+```js
+import * as assets from "hanami-assets";
+
+await assets.run();
+```
+
+If you need more, you can have more! **Assets config can be gracefully extended** to provide advanced esbuild options or [take advantage of its many plugins](https://github.com/esbuild/community-plugins). A fully integrated [PostCSS](https://postcss.org), for example, is just a few lines away:
+
+```js
+import * as assets from "hanami-assets";
+import postcss from "esbuild-postcss";
+
+await assets.run({
+  esbuildOptionsFn: (args, esbuildOptions) => {
+    const plugins = [...esbuildOptions.plugins, postcss()];
+
+    return {
+      ...esbuildOptions,
+      plugins,
+    };
+  },
+});
+```
+
+## Slice it your way
+
+As first-class Hanami features, views and assets work great inside slices as well as your `app/`. Every slice can have its own `views/` and `templates/` directories, for your views, parts, helpers and more.
+
+With Hanami we want to help you draw the right boundaries to support your app’s domain, and views are no different.
+
+## The view ahead looks bright
+
+With Hanami 2.1, we are continuing to deliver our vision of a fresh take for Ruby apps. We’d love for you dive in and give our views and assets a try!
+
+Check out our updated [getting started guide](https://guides.hanamirb.org/v2.1/introduction/getting-started/) for your first steps in building a full stack Hanami app. You’re only a few commands away:
+
+```shell
+$ gem install hanami
+$ hanami new my_app
+$ cd my_app
+$ bundle exec hanami dev
+$ open http://localhost:2300
+```
+
+After 2.1 and views, our next step is the persistence layer. You can look forward to hearing back from us about this early next year.
```

Comment:
Shouldn't this be "... later this year" now?

Reply:
Good pickup! Fixed.